tests/qos/test_buffer.py

```diff
@@ -2994,6 +2995,19 @@ def _check_port_buffer_info_and_return(dut_db_info, table, ids, port, expected_p
         port_config = dut_db_info.get_port_info_from_config_db(port)
         cable_length = cable_length_map[port]
         speed = port_config['speed']
+        port_autoneg = port_config.get('autoneg', "N/A")
+        if port_autoneg == "on":
+            # when autoneg is on, if adv_speeds is configured, use the max adv_speeds to create the profile
```

Comment:
why? what is the logic behind it? the comment does not say anything about that..